### DISCUSSION TITLE
how to get Channel ID from Channel Name Input?

### DISCUSSION THREAD

**Discussion by kholisrag** [slack-go/slack#1026] [Q&A] [answered]
Hi, asking simple and maybe stupid question, I cannot understand after they deprecate the `channel` api, and change it to `conversation` api,
can someone point me how to get channel ID from Channel Name using conversation API or latest slack-go lib?

**Comment by kanata2** [comment]
~~You can use `conversations.info` method.~~ ( I was misunderstanding question)

More details:
- https://pkg.go.dev/github.com/slack-go/slack#Client.GetConversationInfo
- https://api.slack.com/methods/conversations.info

**Comment by kholisrag** [comment, marked as answer]
https://api.slack.com/methods/conversations.info method can used if we known the channel ID,
what I want is input the channel name, and get the channel ID,

anyway fixed with looping method something like this : 
```
	for {
		listChannel, c, err := c.GetConversations(&slack.GetConversationsParameters{
			ExcludeArchived: true,
			Limit:           200,
			Types:           []string{"public_channel", "private_channel"},
			Cursor:          cursor,
		})
		if err != nil {
			logrus.WithField("error_message", err).Error("Unable to Get Channel List")
			break
		}

		if c != "" {
			cursor = c
			logrus.Infof("Searching Next Cursor %v", cursor)

			found := false
			for _, ch := range listChannel {
				if channelName == ch.Name {
					selectedChannel = ch
					found = true
					break
				}
			}
			if found {
				break
			}
			time.Sleep(3 * time.Second)
			continue
		}
	}
```

**Reply by kanata2** [reply]
> what I want is input the channel name, and get the channel ID,

Ah, sorry. I was misunderstanding.
